task/dump-agent/src/udp.rs

```diff
@@ -0,0 +1,167 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{ServerImpl, DUMP_READ_SIZE};
+use hubpack::SerializedSize;
+use ringbuf::*;
+use task_net_api::*;
+
+static_assertions::const_assert_eq!(
+    DUMP_READ_SIZE,
+    humpty::udp::DUMP_READ_SIZE
+);
+
+#[derive(Copy, Clone, PartialEq)]
+enum Trace {
+    None,
+    DeserializeError(hubpack::Error),
+    DeserializeHeaderError(hubpack::Error),
+    SendError(SendError),
+    WrongVersion(u8),
+}
+
+ringbuf!(Trace, 16, Trace::None);
+
+// We are sending a (Header, Result<Response, Error>) to the host
+const MAX_UDP_TX_SIZE: usize = <(
+    humpty::udp::Header,
+    Result<humpty::udp::Response, humpty::udp::Error>,
+)>::MAX_SIZE;
+
+// We are receiving a (Header, Request) from the host
+const MAX_UDP_RX_SIZE: usize =
+    <(humpty::udp::Header, humpty::udp::Request)>::MAX_SIZE;
+
+// Check against packet sizes in the TOML file
+static_assertions::const_assert!(MAX_UDP_TX_SIZE <= 1024);
+static_assertions::const_assert!(MAX_UDP_RX_SIZE <= 1024);
```

Comment:
This might be a bunch of busywork for very little gain so feel free to not do it, but: should our build script emit the socket size so we don't have to keep this literal in sync?

Reply:
Turns out this wasn't too hard, done in 5ceb1a2ab40ae0f8ce5fbf8af1c47b0c60c67c7b